oauth2-authorization-server/src/main/java/org/springframework/security/oauth2/server/authorization/authentication/CodeVerifierAuthenticator.java

```diff
@@ -86,48 +71,53 @@ private boolean authenticate(OAuth2ClientAuthenticationToken clientAuthenticatio
 			throwInvalidGrant(OAuth2ParameterNames.CODE);
 		}
 
-		if (this.logger.isTraceEnabled()) {
-			this.logger.trace("Retrieved authorization with authorization code");
+		if (logger.isTraceEnabled()) {
+			logger.trace("Retrieved authorization with authorization code");
 		}
 
-		OAuth2AuthorizationRequest authorizationRequest = authorization.getAttribute(
+        assert authorization != null;
+        OAuth2AuthorizationRequest authorizationRequest = authorization.getAttribute(
 				OAuth2AuthorizationRequest.class.getName());
+		if (authorizationRequest == null) {
```

Comment:
`authorizationRequest` will never be `null` since `code_verifier` is part of the [Authorization Request](https://www.ietf.org/archive/id/draft-ietf-oauth-v2-1-07.html#section-4.1.1).